### PR TITLE
samples: rtu_server: add support for CDC ACM UART

### LINKED ISSUE #45938: Unable to combine USB CDC-ACM and Modbus Serial due to dependecy on uart_configure().
**Describe the bug**
When combining USB CDC-ACM and Modbus Serial, `modbus_serial_init()` fail because `uart_configure()` isn't implemented for cdc-acm. modbus_serial_init() should handle this more craceful.

Simply ignoring the error if it's -ENOSYS is a quick fix that make modbus serial work.

Testing on nrf52840dk.

**To Reproduce**
1. Enable USB-CDC ACM support with the modbus/rtu_server example.
2. Build, flash and run. uart_configure() returns -ENOSYS since it's not implemented.

**Expected behavior**
Working modbus rtu server over usb cdc-acm.

**Impact**
Need to patch Zephyr code to get it working.

**Logs and console output**
```
[00:00:00.392,028] <err> modbus_serial: Failed to configure UART
[00:00:00.392,028] <err> modbus: Failed to init MODBUS over serial line
[00:00:00.392,059] <err> mbs_sample: Modbus RTU server initialization failed
[00:00:00.396,026] <inf> usb_cdc_acm: Device suspended
[00:00:00.627,502] <inf> usb_cdc_acm: Device resumed
[00:00:00.808,929] <inf> usb_cdc_acm: Device configured
```

**Environment (please complete the following information):**
 - OS: Linux
 - Zephyr SDK
 - Nrf Connect SDK Zephyr repository (v3.0.99-ncs1-rc1). Same issue exist with latest Zephyr commit 70dbf7e695.

### PR DESCRIPTION
**Pull request by jfischer-no**
Allow to use RTU server sample on any board that has supported
USB device controller. Although it is only a point to point
connection and does not represent a bus, it can, apart from
testing the server implementation, also be used practically
for example to control relays or to read ADC values via
USB connection without implementing custom USB class or driver.

Resolves: #45938

### CODE REVIEW

File: samples/subsys/modbus/rtu_server/overlay-cdc-acm.conf [hunk 1]
```diff
@@ -0,0 +1,4 @@
+CONFIG_USB_DEVICE_STACK=y
+CONFIG_USB_DEVICE_PRODUCT="Zephyr CDC ACM MODBUS sample"
+CONFIG_SHELL_BACKEND_SERIAL_CHECK_DTR=y
```
Comment (aurel32) [inline samples/subsys/modbus/rtu_server/overlay-cdc-acm.conf:3]: Why is that needed? The shell is not activated in this sample, causing a warning:

```
warning: SHELL_BACKEND_SERIAL_CHECK_DTR (defined at subsys/shell/backends/Kconfig.backends:75) was
assigned the value 'y' but got the value 'n'. Check these unsatisfied dependencies:
SHELL_BACKEND_SERIAL (=n), SHELL_BACKENDS (=n), SHELL (=n). See
http://docs.zephyrproject.org/latest/kconfig.html#CONFIG_SHELL_BACKEND_SERIAL_CHECK_DTR and/or look
up SHELL_BACKEND_SERIAL_CHECK_DTR in the menuconfig/guiconfig interface. The Application Development
Primer, Setting Configuration Values, and Kconfig - Tips and Best Practices sections of the manual
might be helpful too.
```

**Review by aurel32 — COMMENTED** [review]
Other than that `overlay-cdc-acm.conf` issue, that looks good to me.

**Review by aurel32 — APPROVED** [review]
Thanks for the update, LGTM

**Review by anangl — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]
